### DISCUSSION TITLE
Can I get detailed information about MACCS fingerprint?

### DISCUSSION THREAD

**Discussion by GrailWang** [rdkit/rdkit#6783] [FAQ] [answered]
I have learned that I can get details about certain bits in Mogan fingerprint, like which atom they match and even can draw corresponding structures in figures. But I wonder if I can do the same thing to MACCS fingerprint. I would like to know ,for a certain non-zero bit, where is the matched structure (its index in SMILES) or what exact substructure in the molecule was matched.

**Comment by dehaenw** [comment, marked as answer]
Because each bit's meaning is defined in quite a clear way in MACCS, it should be relatively straightforward to find out what the matching pattern is by just looking up the meaning of the specific bit. You can find the SMARTS pattern used for each MACCS bit in rdkit here: https://github.com/rdkit/rdkit/blob/a9d6642f7dd732ee05814937f0279a05a7485754/rdkit/Chem/MACCSkeys.py#L39

I haven't checked if there's some in built way to visualize the matching fragments, but if not, you can just use the above SMARTS and match it to the index of the on-bits of interest to visualize it.

**Reply by GrailWang** [reply]
Thanks a lot for your answer！
